src/gt4py/next/common.py

```diff
@@ -135,36 +141,128 @@ def __and__(self, other: Set[Any]) -> UnitRange:
         else:
             raise NotImplementedError("Can only find the intersection between UnitRange instances.")
 
+    def __str__(self) -> str:
+        return f"({self.start}:{self.stop})"
 
-DomainRange: TypeAlias = UnitRange | int
+
+def unit_range(r: UnitRangeLike) -> UnitRange:
+    assert is_unit_range_like(r)
```

Comment:
I removed the `UnitRangeLike` as a type completely and just kept `RangeLike`. So unit_range takes `RangelLike` and gives error in case it cannot be interpreted as UnitRange.